src/modal/modal_stack.ts

```diff
@@ -0,0 +1,19 @@
+import {Injectable, TemplateRef} from '@angular/core';
+
+import {NgbModalRef} from './modal_ref';
+
+@Injectable()
+export class NgbModalStack {
+  private modalContainer: any;
```

Comment:
I didn't do it initially since it is creating circular deps between imports. But apparently this is no causing any harm so we've got proper types now.